### ISSUE TITLE
Instruction for install source code

### ISSUE THREAD

**Issue by hungids** [pyronome/pattern-adminlte-3x-vuejs-2x-laravel-6x#1]
Dear author

At first, very thank to you about this source code. 

About me who is a not experience of Laravel and Vuejs, so have some difficult when I install this source code. Can you capture step by step of this source code install.

Thank you

**Comment by aykutaydinli** [comment]
Hello hungids,

Thank you very much for your interest in our repository. Please note that this repository is at the very early stage and under development right now. This repository is a part of a Pyronome -an online source code generator- component. These components called "Patterns". Patterns are used to generate source code in different technologies e.g., Laravel, NodeJS, Reactjs, Vuejs, etc.

To use this pattern, please sign up Pyronome. Pyronome is free. The only limitation is that you have to wait 5 minutes for your next code generation. You can benefit other technologies and features that Pyronome has, such as code generation auto-push to a repository for free.

1. Sign Up / Log-in to Pyronome from https://pyronome.com
2. Create a new project.
3. After creating a new project, a pattern selection dialog will be displayed.
4. Click the "Install" button of the pattern that you want to generate source code. For your case you can choose "AdminLTE 3.x + Vue.js 2.x + Laravel 6.x" pattern.
5. After choosing the pattern. The pattern will be installed to your project. Click the "MODEL" button at the bottom of the pattern selection dialog. The model page will be displayed.
6. On the model page, you need to define your model. These are your classes and your class properties. Source code will be generated based on your model and save you a significant amount of time. You can specify the model below as an example.

```
Student
   FirstName: Short Text
   LastName: Short Text
```

7. After defining your model, click the "Source Code" menu on the right. Your code generation process automatically begins. Your code will be generated in seconds.
8. Download the source code.

Please note that, as I mentioned before, this repository is at a very early stage and under development right now. It is not available for production uses. Just use source code for experimental purposes.

We will be updating these repositories daily. Please check back soon for the latest updates.

**Comment by aykutaydinli** [comment]
Also, I will leave this issue open. I will update the installation steps after downloading the generated source code.

Please feel free to contact me at any time about your questions and suggestions. You can check other repositories as well.

**Comment by obeyfenyere** [comment]
hie

after successfully installed and run the project, the adress http://localhost:8000/adminlte/login returns a blank page
 please help

**Comment by aykutaydinli** [comment]
Hello,

Do you view the default Laravel page when you open  http://localhost:8000/ address?

If you cannot view the default Laravel page, the problem may be the port. Possibly another application/web server is using the port 8000.

Please run the following commands in the terminal:
```
php artisan serve --port=8001
```
and after running the command, please visit the page with the following address:
```
http://localhost:8001/adminlte/login
 ```

**Comment by aykutaydinli** [comment]
Also, please check your model page in the project. Model can be blank. Please define your model and re-generate source code.

**Comment by obeyfenyere** [comment]
http://localhost:8000/ returns the laravel home page...
 
Maybe it is about the model page. let me do that

**Comment by aykutaydinli** [comment]
Ok. Great! Web server is running.
I also installed the source code. It would help if you had the run the following command before running the webserver:

```
npm run dev
```

The command above runs a webpack process and generates required javascript files and copies them into the public directory.

I hope it works for you.

**Comment by obeyfenyere** [comment]
I am new to laravel. i don't know how to define the model page

**Comment by obeyfenyere** [comment]
ERROR in ./resources/js/adminlte/routes.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /home/obey/Projects/source/resources/js/adminlte/routes.js: Unexpected token, expected "," (47:5)

  45 |         "path": ("/" + mainFolder + "/laravelmodeldisplaytext"),
  46 |         "component": AdminLTEModelDisplayTextPage
> 47 |     }{{ROUTES_MODEL_EXPORT_DEFINITONS}}
     |      ^
  48 | ];


getting this error when i run npm run dev

**Comment by aykutaydinli** [comment]
It seems the problem is related to the empty model.
You can define a model in the Pyronome.
Please open your project, and click the "Model" button on the overview page.
You can enter a model using forms on the model page.

If you experience difficulty entering a model, please follow the instructions below:
1. Open your project in the Pyronome.
2. On the "Overview" page, click the "Model" button.
3. Click the JSON link on the right-top of the page.
4. Paste the content of the JSON file [model.json.txt](https://github.com/pyronome/pattern-adminlte-3x-vuejs-2x-laravel-6x/files/5130085/model.json.txt), and save.
5. Close editor page.
6. Click "Source Code" on the menu on the left.
7. Click the "Generate Now!" button.
8. Download the generated source code.
9. Re-install (also run `npm run dev` command, before starting webserver)

Please, also note that this pattern is in the DEVELOPMENT stage. We are updating this pattern daily. Most of the parts can be incomplete and may not work correctly. Use the generated source for experimental purposes.

**Comment by obeyfenyere** [comment]
running the command npm run dev from the main folder doesn't read the package.json file and if i run it from the source folder how will it affect the snippets, install and pyronome folders??

**Comment by aykutaydinli** [comment]
Yes, you are right. Installing command should be run in the `source` folder. Other folders like `pyronome`, `snippets` are the folders necessary in the code generation process (inside the Pyronome). After generating source code, these folders are no effect on the source code.

**Comment by obeyfenyere** [comment]
when i try running php artisan migrate i get the following error

        $adminlte = new App\AdminLTE\AdminLTE();
    26| 
    27|                 $option_data_list = [];
    28| 
  > 29| {{SNIPPET_SEED_MODEL_OPTIONS}}
    30| 
    31|                 $adminlte->seedModelDropdownOptions($option_data_list);
    32|     }}

**Comment by obeyfenyere** [comment]
> when i try running php artisan migrate i get the following error
> 
> ```
>     $adminlte = new App\AdminLTE\AdminLTE();
> 26| 
> 27|                 $option_data_list = [];
> 28| 
> ```
> 
> > 29| {{SNIPPET_SEED_MODEL_OPTIONS}}
> > 30|
> > 31|                 $adminlte->seedModelDropdownOptions($option_data_list);
> > 32|     }}

on line 29

**Comment by aykutaydinli** [comment]
Could you please re-generate the source code, it should complete the migration now.

**Comment by obeyfenyere** [comment]
Thanks!!! all is working fine now

**Comment by aykutaydinli** [comment]
Great! I'm so glad to hear that. As I mentioned before, this pattern is in the development stage. We update this pattern regularly. You need to re-generate source code to benefit from the latest updates.

Please feel free to contact me at any time about your questions and suggestions. You can check other repositories as well.

**Comment by aykutaydinli** [comment]
Also, you can check:
https://pyronome.com/pyronome/adminlte-3x-htmldbjs-1x-laravel-6x

It is way more functional than this one.
In order to use AdminLTE 3.x + HTMLDB.js 1.x + Laravel 6.x pattern, you need to uninstall AdminLTE 3.x + Vue.js 2.x + Laravel 6.x first then install AdminLTE 3.x + HTMLDB.js 1.x + Laravel 6.x from the project's patterns page.

**Comment by obeyfenyere** [comment]
this one still giving the error when migrating database

   25|         $adminlte = new App\AdminLTE\AdminLTE();
    26|
    27|                 $option_data_list = [];
    28|
  > 29| {{SNIPPET_SEED_MODEL_OPTIONS}}
    30|
    31|
$adminlte->seedModelDropdownOptions($option_data_list);
    32|     }}

  Exception trace:

**Comment by aykutaydinli** [comment]
Could you please re-generate source code and re-install it. It should work now.
